### PR TITLE
Add export docs, improve asserts

### PR DESCRIPTION
**Pull request by voznesenskym**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #94961

cc @soumith @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: torch/_dynamo/eval_frame.py [hunk 1]
```diff
@@ -572,7 +602,9 @@ def dynamo_normalization_capturing_compiler(
     ):
         nonlocal graph
 
-        assert graph is None, "whole graph export entails exactly one graph"
+        assert (
+            graph is not None
```
Comment (qihqi) [inline torch/_dynamo/eval_frame.py:606]: old code says `graph is None`...

**Review by qihqi — COMMENTED** [review]
what we gonna do with the input/output comment?

File: torch/_dynamo/eval_frame.py (hunk 1 above)
Comment (voznesenskym) [reply torch/_dynamo/eval_frame.py:606]: bad copy pastes, test caught it lol

**Comment by ezyang** [comment]
I'm going to let the actual users assess the usefulness of these docs

File: torch/_dynamo/eval_frame.py [hunk 2]
```diff
@@ -521,6 +521,36 @@ def guard_export_print(guards):
...
+
+    Args:
+        f (callable): A PyTorch function to be exported.
+
+        *args: Variable length argument list to be passed to the function f.
+
+        aten_graph (bool): If True, exports a graph with ATen operators.
+        If False, exports a graph with Python operators. Default is False.
+
+        decomposition_table (dict): A dictionary that maps operators to their decomposition functions.
```
Comment (msaroufim) [inline torch/_dynamo/eval_frame.py:535]: Commenting as a humble end user: Not sure here what a decomposition table concretely looks like beyond a dict and what's teh default if I don't pass anything

not sure what dual and real mean exactly in tracing mode

The return type last I checked maybe a week ago was a tuple with a GraphModule and Guards, if that changed recently ignore this line

**Review by tugsbayasgalan — APPROVED** [review]

File: torch/_dynamo/eval_frame.py [hunk 3]
```diff
@@ -521,6 +521,36 @@ def guard_export_print(guards):
...
+        Required if aten_graph or tracing_mode is specified. Default is None.
+
+        tracing_mode (str): Specifies the tracing mode. Must be set to "real" if decomposition_table is not specified.
```
Comment (tugsbayasgalan) [inline torch/_dynamo/eval_frame.py:538]: What does dual do?

Comment (voznesenskym) [reply torch/_dynamo/eval_frame.py:538]: It should say fake, I think I accidentally a word with my touchbar

**Pull request closed without merging**